### DISCUSSION TITLE
Installation other than aspx pages

### DISCUSSION THREAD

**Discussion by sb-sheriff** [microsoft-search/pnp-modern-search#1459] [Q&A] [answered]
Hi,

We could install pnp search by adding it via web part on the standard SP home page and replace the Microsoft built-in search that way. But how to utilize pnp search for other than aspx pages? Such as the document library page, which has the built-in search and does not have a way to install web part nor pnp search.

Thank you!

**Comment by wobba** [comment, marked as answer]
Not possible. Web parts can only be added on web part pages. Why do you want to replace search for a document library, as that is more of a filter experience?

**Reply by sb-sheriff** [reply]
The reason to replace the built-in search box is the same we did on the home page -- the built-in search doesn't return for the exact search result and the better filter experience as well.

With that said, the pnp search will work along with the built-in search together on SP sites then? It would be confusing to users that they don't actually work and return the same results.

**Reply by wobba** [reply]
Can you elaborate on this? "the built-in search doesn't return for the exact search result". And filters are coming for site search. And for libraries filtering and searching on the columns is indeed a good use case.

**Comment by sb-sheriff** [comment]
The Microsoft built-in search couldn't return the exact search result as the user requested, which I had opened the previous issue on it:
https://github.com/microsoft-search/pnp-modern-search/issues/1146

Simply put, the search on the pages other than the home page (with pnp installed) behave differently is confusing to the users and even developers.
